### PR TITLE
Introduce interfaces for service dependencies and span storage.

### PR DESCRIPTION
**Pull request by badiib**

### CODE REVIEW

File: storage/README.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+collection of different storage interfaces that are shared by two or more components.
```
Comment (badiib) [inline storage/README.md]: Should we rewrite to:

A collection of different storage interfaces that are shared by two or more components

Comment (black-adder) [inline storage/README.md]: proper english is good english, make sure you update the comment in storage/doc.go as well.

Comment (yurishkuro) [inline storage/README.md]: this was my attempt to use READMEs (which are handy on github) to generate `doc.go` with md-to-godoc tool. But the tool has this difficulty with figuring out the first sentence.

File: storage/spanstore/metrics/decorator_test.go [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+		"GetTrace.errors":         1,
+		"FindTraces.attempts":     1,
+		"FindTraces.successes":    0,
+		"FindTraces.errors":       1,
+		"GetServices.attempts":    1,
+		"GetServices.successes":   0,
+		"GetServices.errors":      1,
+	}
+
+	for k, v := range expecteds {
```
Comment (yurishkuro) [inline storage/spanstore/metrics/decorator_test.go]: from here down looks like it could be DRY-ed out with the previous test.

**Pull request merged**